.golangci.yml

```diff
@@ -0,0 +1,36 @@
+linters-settings:
+  govet:
+    check-shadowing: true
+
+linters:
+  disable-all: true
+  enable:
+    - unused
+    - gosimple
+    - structcheck
+    - varcheck
+    - ineffassign
+    - deadcode
+    - unconvert
+    # TODO: enable these later
+    # - govet
+    # - gofmt
```

Comment:
Because of `new-from-rev`. If we use govet and gofmt from golangci, then the separate govet and gofmt is redundant. And we cannot enable it here because it will only check the current commit. But that's moot if we are using golangci service. I will enable them.